### DISCUSSION TITLE
How to retrieve ID of guild bot is currently in?

### DISCUSSION THREAD

**Discussion by telephrag** [bwmarrin/discordgo#1234] [Q&A] [answered]
I'm writing a single-guild self hosted bot. In other words bot shall be disallowed to be in more than one guild at once and user will have to host bot on his own instead of "inviting" bot that is already hosted somewhere to the server. 

To save a hustle for end user I want to get guild id at runtime so, that user didn't have to do it manually. Also, I'll need guild ids to check on startup, if bot is already present in some guild.

**Comment by telephrag** [comment]
Anyway, I've made a PR: https://github.com/bwmarrin/discordgo/pull/1235

**Comment by FedorLap2006** [comment, marked as answer]
To get all guilds bot is currently in, you can use `Session.State.Ready.Guilds` (or just `Session.State.Guild`).
However I would recommend listening for `GuildCreate` event, which is fired each time bot joins a new guild.

P.S. Not really sure why you're trying to implement this by yourself. Since the same functional can be achieved by toggling "Public bot" option in Developer Dashboard.

**Reply by telephrag** [reply]
Does toggling public bot prevents bot from joining multiple guilds? 
UPD: No, it doesn't. I want to prevent a bot usage in multiple servers.

**Reply by FedorLap2006** [reply]
It does not prevent bot from joining multiple guilds. But only the bot owner would be able to invite it somewhere.
If you spefcifically need to lock it into a single guild, the only way is to add checks yourself.

**Reply by telephrag** [reply]
Yep, that's exactly what I did. Just to be sure.
